iceoryx_posh/test/moduletests/test_popo_listener_waitset_attachments.cpp

```diff
@@ -0,0 +1,210 @@
+/// are using the same trigger concept
+class ListenerWaitsetAttachments_test : public Test
+{
+  public:
+    void SetUp()
+    {
+        EXPECT_CALL(*this->runtimeMock, getMiddlewareConditionVariable())
+            .WillOnce(Return(&this->conditionVariableData));
+        listener.emplace();
+    }
+
+    template <typename T>
+    static void genericTriggerCallback(T* const)
+    {
+    }
+
+    std::unique_ptr<PoshRuntimeMock> runtimeMock = PoshRuntimeMock::create(RUNTIME_NAME);
+    ConditionVariableData conditionVariableData{RUNTIME_NAME};
+    optional<Listener> listener;
+
+    MemoryManager memoryManager;
+};
+
+
+TEST_F(ListenerWaitsetAttachments_test, SubscriberAttachesToListener)
+{
+    ::testing::Test::RecordProperty("TEST_ID", "f1d163e0-4479-47b4-b4e8-01b0ee6a71b0");
+    SubscriberPortData subscriberData({SERVICE, INSTANCE, EVENT},
+                                      RUNTIME_NAME,
+                                      VariantQueueTypes::SoFi_MultiProducerSingleConsumer,
+                                      SubscriberOptions());
+    EXPECT_CALL(*this->runtimeMock, getMiddlewareSubscriber(_, _, _)).WillOnce(Return(&subscriberData));
+
+    optional<Subscriber<int>> subscriber;
+    subscriber.emplace(ServiceDescription(SERVICE, INSTANCE, EVENT));
+
+    ASSERT_FALSE(listener
+                     ->attachEvent(*subscriber,
+                                   SubscriberEvent::DATA_RECEIVED,
+                                   createNotificationCallback(
+                                       ListenerWaitsetAttachments_test::genericTriggerCallback<Subscriber<int>>))
+                     .has_error());
+
+    EXPECT_THAT(listener->size(), Eq(1));
+    subscriber.reset();
+    EXPECT_THAT(listener->size(), Eq(0));
+}
+
+TEST_F(ListenerWaitsetAttachments_test, UntypedSubscriberAttachesToListener)
+{
+    ::testing::Test::RecordProperty("TEST_ID", "a78a7016-46b6-4223-b7b1-e30344bb208f");
+    SubscriberPortData subscriberData({SERVICE, INSTANCE, EVENT},
+                                      RUNTIME_NAME,
+                                      VariantQueueTypes::SoFi_MultiProducerSingleConsumer,
+                                      SubscriberOptions());
+    EXPECT_CALL(*this->runtimeMock, getMiddlewareSubscriber(_, _, _)).WillOnce(Return(&subscriberData));
+
+    optional<UntypedSubscriber> subscriber;
+    subscriber.emplace(ServiceDescription(SERVICE, INSTANCE, EVENT));
+
+    ASSERT_FALSE(listener
+                     ->attachEvent(*subscriber,
+                                   SubscriberEvent::DATA_RECEIVED,
+                                   createNotificationCallback(
+                                       ListenerWaitsetAttachments_test::genericTriggerCallback<UntypedSubscriber>))
+                     .has_error());
+
+    EXPECT_THAT(listener->size(), Eq(1));
+    subscriber.reset();
+    EXPECT_THAT(listener->size(), Eq(0));
+}
+
+TEST_F(ListenerWaitsetAttachments_test, ClientAttachesToListener)
```

Comment:
Concise test case name, describing action and result. Same for the rest of the test cases.